cpp/include/cudf/lists/detail/scatter.cuh

```diff
@@ -0,0 +1,727 @@
+                                    // to be written for this list row_index.
+      auto input_list_start =
+        lists_offsets_column.template element<int32_t>(unbound_list_view.row_index());
+
+      thrust::for_each_n(
+        thrust::seq,
+        thrust::make_counting_iterator<size_type>(0),
+        actual_list_row.size(),
+        [output_start_offset,
+         d_string_views,
+         input_list_start,
+         d_string_offsets = string_offsets_column.template data<int32_t>(),
+         d_string_chars =
+           string_chars_column.template data<char>()] __device__(auto const& string_idx) {
+          // auto string_offset     = output_start_offset + string_idx;
+          auto string_start_idx = d_string_offsets[input_list_start + string_idx];
+          auto string_end_idx   = d_string_offsets[input_list_start + string_idx + 1];
+
+          d_string_views[output_start_offset + string_idx] =
+            string_view{d_string_chars + string_start_idx, string_end_idx - string_start_idx};
+        });
+    };
+
+    thrust::for_each_n(rmm::exec_policy(stream)->on(stream),
+                       thrust::make_counting_iterator<size_type>(0),
+                       list_vector.size(),
+                       populate_string_views);
+
+    // string_views should now have been populated with source and target references.
+
+    auto string_offsets =
+      cudf::strings::detail::child_offsets_from_string_vector(string_views, mr, stream);
+    auto string_chars = cudf::strings::detail::child_chars_from_string_vector(
+      string_views, string_offsets->view().data<int32_t>(), 0, mr, stream);
+    auto child_null_mask =
+      source_lists_column_view.child().nullable() || target_lists_column_view.child().nullable()
+        ? construct_child_nullmask(
+            list_vector, list_offsets, source_lists, target_lists, num_child_rows, mr, stream)
+        : std::make_pair(rmm::device_buffer{}, 0);
+
+    return cudf::make_strings_column(num_child_rows,
+                                     std::move(string_offsets),
+                                     std::move(string_chars),
+                                     child_null_mask.second,            // Null count.
+                                     std::move(child_null_mask.first),  // Null mask.
+                                     stream,
+                                     mr);
+  }
+
+  /**
+   * @brief (Recursively) Constructs a child column that is itself a list column.
+   */
+  template <typename T>
+  std::enable_if_t<std::is_same<T, list_view>::value, std::unique_ptr<column>> operator()(
+    rmm::device_vector<unbound_list_view> const& list_vector,
+    cudf::column_view const& list_offsets,
+    cudf::lists_column_view const& source_lists_column_view,
+    cudf::lists_column_view const& target_lists_column_view,
+    rmm::mr::device_memory_resource* mr,
+    cudaStream_t stream) const
```

Comment:
`rmm::cuda_stream_view`, and reorder to stream, mr.

Reply:
Fixed. Will post update shortly.